src/plugins/discover/public/application/angular/discover.js

```diff
@@ -972,6 +1010,10 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
       )
       .setField('query', data.query.queryString.getQuery() || null)
       .setField('filter', filterManager.getFilters());
+    if (useNewFieldsApi) {
+      searchSource.setField('fields', ['*']);
```

Comment:
If we don't request all fields, what about:
1. default view?
2. how do we add a new column? (see discussion here: https://github.com/elastic/kibana/pull/75407/files#r473136754)

Reply:
Wouldn't we, if `useNewFieldsApi` was set to false, request `_source` in the legacy way? so it would function like before?

Reply:
That's true, but my observation was rather around usage of the new fields API. Ideally, we would want to request only the fields specified in the data grid. If we do that, this leads us to the problems specified above. Unless what you're saying is we should still request `_source ` and make this work as before?

Reply:
I do agree, would also prefer to request what is needed, this is a better pattern. Why going to the supermarket buying all the groceries you could eventually need, your bags might get heavy. However, in this case I would as a first step keep the legacy mode in the same way it's currently used, to prevent adding additional complexity we were talking about https://github.com/elastic/kibana/pull/75407/files#r473136754). Could be done later on